text/0000-crate-changelog-field.md

```diff
@@ -0,0 +1,228 @@
+- This proposal does not constrain the format of the changelog, making it hard
+  for external tools to consume the contents of the changelog.
+
+# Rationale and Alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Rationale
+[rationale]: #rationale
+
+Most of the time, the producer of changelogs are crate authors
+and the consumer of changelogs are other crate authors.
+For human consumption, it is ideal to have the changelog in a rendered format.
+But handling rendering of various formats in crates.io is a heavy task.
+However, many existing crates already have changelogs available in their
+repository and the websites backing their repositories can already render the
+changelogs. So, in order to avoid unnecessary efforts, this RFC proposes to
+just use a valid URL for the `changelog` field.
+
+## Alternative: Local File
+[alternative-local-file]: #alternative-local-file
+
+The `changelog` field can be a path to a file within the crate,
+which opens the possibility for crates.io to render it directly.
+However, people have different opinions when it comes to what format the
+changelog should use. While markdown is currently the most popular choice,
+plain text and reStructuredText are also in use.
+Instead of handling the rendering of various different formats of changelog in
+crates.io, it is much easier to just have an external link, where an external
+website handles the rendering of the changelog so that crate authors can use
+any format they see fit for their changelogs.
+
+And changelogs grow over time, which can take significant amount of disk spaces
+for crates.io, considering that usually every single release of a crate
+will contain all changelogs up to that release.
+This particular drawback can be solved by introducing new linting rules.
+
+From another perspective, using an external link can guarantee that we can
+always link to the latest changelog, which is supposed to contain all
+changelog entries from initial versions to unreleased commits.
+If we adopt the local file approach, when we view an outdated version of a
+crate on docs.rs, we only get a link to the outdated changelog, which is not
+helpful for viewing the breaking changes introduced in new versions, requring
+another click on the doc.rs page to switch to the latest version.
+
+## Alternative: Do Nothing
+[alternative-do-nothing]: #alternative-do-nothing
+
+It has been several years after the initial changelog RFC.
+We can continue to live without this enhancement.
+
+It is possible to create an external registry that collects the links of
+changelogs. However, such effort is neither scalable nor ideal.
+
+# Prior art
+[prior-art]: #prior-art
+
+[Previously closed RFC](https://github.com/rust-lang/rfcs/pull/2129)
+proposed to use a strictly formatted Markdown file or an external link for the
+changelog field. The strict format, while ideal for machine consumption,
+imposes too much constraint and lead to a refusal.
```

Comment:
This belongs in Rationale / Alternatives

Reply:
Done in 4e99d945